engineering-decision-records/026-tracing-span-design.md

```
@@ -18,116 +20,179 @@ This section is structured by the kind of traces under consideration:
+[EDR-007 on observability][edr-observability] defines the general approach to observability and the use of
+OpenTelemetry. It prescribes the use of a schema to specify spans attributes and their types with namespacing to avoid
+collisions.
+
+The current EDR restricts the definition of the schema to five levels of nesting, where the first two levels are used to
+define the target of a span and the next 3 levels to define the name of the span.
+
+Here are two examples. During consensus, we receive an upstream message informing the node of a new block header and we
+need
+to decode it. In that case the fully qualified name of the span in the schema is:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)